Subject: [PATCH] Ensure the configuration of OWNER completely parsed.

The contents of /data/system/users/0/0.xml might be damaged in power-fail case.
PackageManager would throw RuntimeException if no system user are found,
resulting zygote restarting.

userlist.xml and 0.xml in /data/system/users must be synchronized.
If the 0.xml is missing or damaged, redit both of them.

diff --git a/services/core/java/com/android/server/pm/UserManagerService.java b/services/core/java/com/android/server/pm/UserManagerService.java
index 4faa506..f7b7d4c 100644
--- a/services/core/java/com/android/server/pm/UserManagerService.java
+++ b/services/core/java/com/android/server/pm/UserManagerService.java
@@ -1735,6 +1735,9 @@
             }
             updateUserIds();
             upgradeIfNecessaryLP();
+            if (mUsers.get(UserHandle.USER_SYSTEM) == null) {
+                fallbackToSingleUserLP();
+            }
         } catch (IOException | XmlPullParserException e) {
             fallbackToSingleUserLP();
         } finally {
